hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncScanSingleRegionRpcRetryingCaller.java

```diff
@@ -420,10 +422,17 @@ private void onError(Throwable error) {
     }
     long delayNs;
     long pauseNsToUse =
-      HBaseServerException.isServerOverloaded(error) ? pauseNsForServerOverloaded : pauseNs;
+      HBaseServerExceptionPauseManager.getPauseNanos(error, pauseNsForServerOverloaded, pauseNs);
+    if (
+      error instanceof RpcThrottlingException && pauseNsToUse > remainingTimeNs() - SLEEP_DELTA_NS
+    ) {
+      completeExceptionally(!scannerClosed);
+      return;
+    }
     if (scanTimeoutNs > 0) {
-      long maxDelayNs = remainingTimeNs() - SLEEP_DELTA_NS;
-      if (maxDelayNs <= 0) {
+      long remainingTimeNs = remainingTimeNs();
+      long maxDelayNs = remainingTimeNs - SLEEP_DELTA_NS;
+      if (maxDelayNs <= 0 || maxDelayNs <= pauseNsToUse) {
```

Comment:
trying to parse this change... i feel like we can ditch this diff, since we handle the completeExceptionally for throttling above? That was the only change I wanted here, not sure if this is accomplishing something else.

Reply:
Yeah you're right